gst/nnstreamer/tensor_query/tensor_query_common.c

```diff
@@ -620,25 +651,127 @@ query_tcp_send (GSocket * socket, uint8_t * data, size_t size,
+  if (0 != nnstreamer_query_receive (_conn, &cmd_data, 1)) {
+    nns_logi ("Failed to receive cmd");
+    goto done;
+  }
+
+  if (cmd_data.cmd == _TENSOR_QUERY_CMD_REQUEST_INFO) {
+    GstTensorsConfig *config = &cmd_data.data_info.config;
+    if ((gst_tensors_config_is_flexible (config) &&
+            gst_tensors_config_is_flexible (&_sdata->src_config)) ||
+        gst_tensors_info_is_equal (&config->info, &_sdata->src_config.info)) {
+      cmd_data.cmd = _TENSOR_QUERY_CMD_RESPOND_APPROVE;
+      /* respond sink config */
+      gst_tensors_config_copy (config, &_sdata->sink_config);
+    } else {
+      /* respond deny with src config */
+      nns_logw ("tensor info is not equal");
+      cmd_data.cmd = _TENSOR_QUERY_CMD_RESPOND_DENY;
+      gst_tensors_config_copy (config, &_sdata->src_config);
+    }
+    if (nnstreamer_query_send (_conn, &cmd_data, DEFAULT_TIMEOUT_MS) != 0) {
+      nns_logi ("Failed to send respond");
+      goto done;
+    }
+  }
+
+  while (_conn->running) {
+    condition = g_socket_condition_check (_conn->socket,
+        G_IO_IN | G_IO_PRI | G_IO_ERR | G_IO_HUP);
+
+    size = g_socket_get_available_bytes (_conn->socket);
+
+    if (condition && size <= 0) {
+      nns_loge ("socket not available, possibly EOS");
+      break;
+    }
+
+    if (0 != nnstreamer_query_receive (_conn, &cmd_data, 1)) {
+      nns_loge ("Failed to receive cmd");
+      break;
+    }
+
+    if (cmd_data.cmd == _TENSOR_QUERY_CMD_TRANSFER_START) {
+      data_info = cmd_data.data_info;
+      outbuf = gst_buffer_new ();
+      for (i = 0; i < data_info.num_mems; i++) {
+        mem = gst_allocator_alloc (NULL, data_info.mem_sizes[i], NULL);
+        gst_buffer_append_memory (outbuf, mem);
+
+        if (!gst_memory_map (mem, &map, GST_MAP_READWRITE)) {
+          nns_loge ("Failed to map the memory to receive data.");
+          goto reset_buffer;
+        }
+
+        cmd_data.data.data = map.data;
+        cmd_data.data.size = map.size;
+
+        ecode = nnstreamer_query_receive (_conn, &cmd_data, 1);
+        gst_memory_unmap (mem, &map);
+
+        if (ecode != 0) {
+          nns_logi ("Failed to receive data");
+          goto reset_buffer;
+        }
+      }
+
+      /* receive end */
+      if (0 != nnstreamer_query_receive (_conn, &cmd_data, 1) ||
+          cmd_data.cmd != _TENSOR_QUERY_CMD_TRANSFER_END) {
+        nns_logi ("Failed to receive end command");
+        goto reset_buffer;
+      }
+
+      meta_query = gst_buffer_add_meta_query (outbuf);
+      if (meta_query) {
+        meta_query->client_id =
+            nnstreamer_query_connection_get_client_id (_conn);
+      }
+    } else {
+      nns_logi ("Failed to receive start command.");
+      break;
     }
-    g_async_queue_push (queue, conn);
+    g_async_queue_push (_sdata->msg_queue, outbuf);
+    continue;
+  reset_buffer:
+    gst_buffer_unref (outbuf);
   }
 
-  return client_id;
+done:
+  g_free (thread_data);
+  thread_data = NULL;
```

Comment:
`thread_data` is an argument so this assignment is meaningless..

Reply:
Right. Updated.